### ISSUE TITLE
Test Case for Enum Types

### ISSUE THREAD

**Issue by tonymichaelson** [scala-tsi/scala-tsi#74]
Hello,

Can a test case please be created to verify enums are properly generated in Typescript output from the serializer?

I have a simple case class with an enum type for one of the fields. I would like a test case to refer to for proper usage of your library for this situation.

Many thanks,
Tony

**Comment by dhoepelman** [comment]
Enums are indeed not support out-of-the-box yet, It's a good addition.

In the mean time, assuming you mean [Scala Enumeration](https://www.scala-lang.org/api/current/scala/Enumeration.html), something like this should work:

```
def enumeration[E <: Enumeration](name: String, enum: E): TSType[E] =
  TSType.alias[E](name, TSUnion(enum.values.iterator.map(entry => TSLiteralString(entry)).toSeq))

implicit val myEnumTsType: TSType[MyEnum] = enumeration("MyEnum", MyEnum)
``` 

This will result in:
```
type MyEnum = "EntryA" | "EntryB" | "EntryC" | ...
```


If you use the excellent [enumeratum](https://github.com/lloydmeta/enumeratum) library, you can use this:
```scala
def enumeratum[AllValues <: Enum[E]: Manifest, E <: EnumEntry](enum: AllValues)(implicit m: Manifest[AllValues]): TSType[E] =
    TSType.alias[E](
      m.runtimeClass.getSimpleName.stripSuffix("$"),
      TSUnion(enum.values.map(e => TSLiteralString(e.entryName)))
    )

implicit val myEnum: TSType[MyEnum] = enumeratum(MyEnum)
```

**Comment by tonymichaelson** [comment]
Thank you for the assistance David, I was using Scala Enumeration but failed to get that working. I switched to enumeratum and modified your solution a bit to compile. Including my enum object, for future reference, the following code is working:


```scala
scalaVersion := "2.13.2"
"nl.codestar"  % "sbt-scala-tsi"             % "0.2.2"
"com.beachape" % "enumeratum-play-json_2.13" % "1.5.17"
```

```scala
import enumeratum.{PlayJsonEnum, Enum, EnumEntry}

sealed trait MapMode extends EnumEntry
object MapMode extends Enum[MapMode] with PlayJsonEnum[MapMode] {
  val values = findValues
  case object MAP      extends MapMode
  case object LIST     extends MapMode
  case object DOCUMENT extends MapMode
}

  def enumTsType[E <: enumeratum.EnumEntry, AllValues <: enumeratum.Enum[E]: Manifest](enum: AllValues)(
      implicit m: Manifest[AllValues]): TSType[E] =
    TSType.alias[E](
      m.runtimeClass.getSimpleName.stripSuffix("$"),
      TSUnion(enum.values.map(e => TSLiteralString(e.entryName)))
    )
  implicit val myEnumTsType: TSType[MapMode] = enumTsType(MapMode)
```

Directly adding the `mapMode` enum shown above to the `typescriptClassesToGenerateFor` SBT key results in a funky code generation of:

```typescript
export interface IDOCUMENT {}
export interface ILIST {}
export interface IMAP {}
export type MapMode = (IDOCUMENT | ILIST | IMAP)
```

but if the enum is simply referenced from a case class that is in the SBT key instead, the code generated is a lot more desirable for me:

```typescript
export type MapMode = ("MAP" | "LIST" | "DOCUMENT")
```

Thank you again for helping, I'm very happy to have this working in my project before the end of this week now :)

Cheers,
Tony

**Comment by dhoepelman** [comment]
@tonymichaelson Glad to hear!

> Directly adding the mapMode enum shown above to the typescriptClassesToGenerateFor SBT key results in a funky code generation of

What was the `build.sbt` that results in the funky code generation? I'd like to improve that if possible

**Comment by tonymichaelson** [comment]
Awesome! I saw your latest release, thank you so much!

I've created a project based on my use case: https://github.com/tonymichaelson/scala-tsi-issue-74

I'm exporting the enum directly via `typescriptClassesToGenerateFor` and also as a field referenced from a case class. The case class reference produces the desired Typescript with string literals but the direct export does not.

I also noticed if I use `typescriptClassesToGenerateFor` *and also* reference the enum from a case class it will be exported *twice*. I added that to the project as well. I hope it's clear :)

Thanks again,
Tony
